<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindowConstrainedIKDemo</class>
 <widget class="QMainWindow" name="MainWindowConstrainedIKDemo">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>819</width>
    <height>945</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>IK Demo</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QFrame" name="frameViewer">
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
      </widget>
      <widget class="QGroupBox" name="groupBox">
       <property name="maximumSize">
        <size>
         <width>200</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="title">
        <string/>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QPushButton" name="pushButtonReset">
          <property name="text">
           <string>Reset</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButtonLoad">
          <property name="text">
           <string>(Re-)Load</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_3">
          <property name="title">
           <string>Kinematic Chain</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QComboBox" name="comboBoxKC"/>
           </item>
           <item>
            <widget class="QLabel" name="label_NrJoints">
             <property name="text">
              <string>Nr of joints:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_TCP">
             <property name="text">
              <string>TCP:</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="poseGroup">
          <property name="title">
           <string>Pose Constraint</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="checked">
           <bool>false</bool>
          </property>
          <layout class="QFormLayout" name="formLayout">
           <item row="3" column="0">
            <widget class="QLabel" name="label_11">
             <property name="text">
              <string>y</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0" colspan="2">
            <widget class="QPushButton" name="poseRandom">
             <property name="text">
              <string>Random</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_10">
             <property name="text">
              <string>x</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QDoubleSpinBox" name="poseX">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimum">
              <double>-10000.000000000000000</double>
             </property>
             <property name="maximum">
              <double>10000.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QDoubleSpinBox" name="poseY">
             <property name="minimum">
              <double>-10000.000000000000000</double>
             </property>
             <property name="maximum">
              <double>10000.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="6" column="0">
            <widget class="QLabel" name="label_8">
             <property name="text">
              <string>z</string>
             </property>
            </widget>
           </item>
           <item row="6" column="1">
            <widget class="QDoubleSpinBox" name="poseZ">
             <property name="minimum">
              <double>-10000.000000000000000</double>
             </property>
             <property name="maximum">
              <double>10000.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="8" column="0">
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>roll</string>
             </property>
            </widget>
           </item>
           <item row="8" column="1">
            <widget class="QDoubleSpinBox" name="poseRoll">
             <property name="minimum">
              <double>-3.140000000000000</double>
             </property>
             <property name="maximum">
              <double>3.140000000000000</double>
             </property>
            </widget>
           </item>
           <item row="10" column="0">
            <widget class="QLabel" name="label_12">
             <property name="text">
              <string>pitch</string>
             </property>
            </widget>
           </item>
           <item row="10" column="1">
            <widget class="QDoubleSpinBox" name="posePitch">
             <property name="minimum">
              <double>-3.140000000000000</double>
             </property>
             <property name="maximum">
              <double>3.140000000000000</double>
             </property>
            </widget>
           </item>
           <item row="12" column="0">
            <widget class="QLabel" name="label_9">
             <property name="text">
              <string>yaw</string>
             </property>
            </widget>
           </item>
           <item row="12" column="1">
            <widget class="QDoubleSpinBox" name="poseYaw">
             <property name="minimum">
              <double>-3.140000000000000</double>
             </property>
             <property name="maximum">
              <double>3.140000000000000</double>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="tsrGroup">
          <property name="title">
           <string>TSR &amp;Constraint</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="checked">
           <bool>false</bool>
          </property>
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="1" column="3">
            <widget class="QDoubleSpinBox" name="tsrHighX">
             <property name="minimum">
              <double>-10000.000000000000000</double>
             </property>
             <property name="maximum">
              <double>10000.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="6" column="1">
            <widget class="QLabel" name="label_6">
             <property name="text">
              <string>yaw</string>
             </property>
            </widget>
           </item>
           <item row="6" column="3">
            <widget class="QDoubleSpinBox" name="tsrHighYaw">
             <property name="minimum">
              <double>-3.140000000000000</double>
             </property>
             <property name="maximum">
              <double>3.140000000000000</double>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>y</string>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QDoubleSpinBox" name="tsrLowX">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimum">
              <double>-10000.000000000000000</double>
             </property>
             <property name="maximum">
              <double>10000.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="3" column="2">
            <widget class="QDoubleSpinBox" name="tsrLowZ">
             <property name="minimum">
              <double>-10000.000000000000000</double>
             </property>
             <property name="maximum">
              <double>10000.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="5" column="3">
            <widget class="QDoubleSpinBox" name="tsrHighPitch"/>
           </item>
           <item row="6" column="2">
            <widget class="QDoubleSpinBox" name="tsrLowYaw">
             <property name="minimum">
              <double>-3.140000000000000</double>
             </property>
             <property name="maximum">
              <double>3.140000000000000</double>
             </property>
            </widget>
           </item>
           <item row="4" column="2">
            <widget class="QDoubleSpinBox" name="tsrLowRoll">
             <property name="minimum">
              <double>-3.140000000000000</double>
             </property>
             <property name="maximum">
              <double>3.140000000000000</double>
             </property>
            </widget>
           </item>
           <item row="5" column="2">
            <widget class="QDoubleSpinBox" name="tsrLowPitch">
             <property name="minimum">
              <double>-3.140000000000000</double>
             </property>
             <property name="maximum">
              <double>3.140000000000000</double>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>x</string>
             </property>
            </widget>
           </item>
           <item row="4" column="3">
            <widget class="QDoubleSpinBox" name="tsrHighRoll">
             <property name="minimum">
              <double>-3.140000000000000</double>
             </property>
             <property name="maximum">
              <double>3.140000000000000</double>
             </property>
            </widget>
           </item>
           <item row="2" column="3">
            <widget class="QDoubleSpinBox" name="tsrHighY">
             <property name="minimum">
              <double>-10000.000000000000000</double>
             </property>
             <property name="maximum">
              <double>10000.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>z</string>
             </property>
            </widget>
           </item>
           <item row="2" column="2">
            <widget class="QDoubleSpinBox" name="tsrLowY">
             <property name="minimum">
              <double>-10000.000000000000000</double>
             </property>
             <property name="maximum">
              <double>10000.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="5" column="1">
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>pitch</string>
             </property>
            </widget>
           </item>
           <item row="3" column="3">
            <widget class="QDoubleSpinBox" name="tsrHighZ">
             <property name="minimum">
              <double>-10000.000000000000000</double>
             </property>
             <property name="maximum">
              <double>10000.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>roll</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0" colspan="4">
            <widget class="QPushButton" name="tsrRandom">
             <property name="text">
              <string>Random</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButtonSolve">
          <property name="text">
           <string>Solve IK</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_6">
          <property name="title">
           <string>Result</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <item>
            <widget class="QLabel" name="labelTime">
             <property name="text">
              <string>Time: 0 ms</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="labelPos">
             <property name="text">
              <string>Error Pos: 0 mm</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="labelOri">
             <property name="text">
              <string>Error Ori: 0 deg</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>819</width>
     <height>20</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
